### DISCUSSION TITLE
Kanidm RADIUS SSO

### DISCUSSION THREAD

**Discussion by Jburso** [kanidm/kanidm#3263] [General]
My understanding of how RADIUS credentials are managed in Kanidm is that credentials can only be generated. They cannot be set by the user, unlike the Kanidm and UNIX credentials. As a result, the generated password must be stored or remembered by the user in addition to their other passwords.

Are there any plans to allow users to set the RADIUS credential, or even enable the use of the UNIX credential (similar to how Kanidm LDAP works)? This would allow for a better SSO experience for the user.

**Comment by yaleman** [comment]
No current plans, and if we were to make changes I'd say it would be to move all RADIUS auth entirely to mTLS instead of passwords.

**Reply by Gorian** [reply]
FYI, if you make this change, please keep the ability to use passwords - I agree that mTLS is more secure, and it's likely the best option in the scenario that RADIUS is being used to authenticate Wifi connections.

However, in the scenario where RADIUS is being used to add SSO to network devices like switches and routers that you would connect to over SSH for example - I don't think that mTLS is a supported method.

**Reply by Firstyear** [reply]
We have no plans to remove RADIUS PEAP-MSCHAPv2 password support as it's "universally accepted". mTLS is just a "more secure option in addition".

**Reply by Gorian** [reply]
Thanks for the confirmation FirstYear :) I definitely appreciate the work you put into security first.

**Comment by Firstyear** [comment]
@Jburso The issue is RADIUS credentials are insecure - both from a protocol level within RADIUS, but also how they have to be stored to support NTLM/MSCHAPv2 which is the only "standard" RADIUS auth mechanism that "just works" on all platforms.

If we allowed users to set passwords, then users would set their password to be the same as their account password *or similar* to it. Since RADIUS necessitates and forces insecure credential storage, this means we would be storing the users password effectively as plaintext. This then means that the blast radius of a compromised credential expands rapidly. 

As a result, we made the choice that radius passwords *must* be generated separately to the main account password to defend from this. This way the RADIUS password has a single purpose (auth to RADIUS) and compromise of it will never compromise anything (such as SSH auth to a machine or loging into Kanidm itself). 

This does mean that RADIUS passwords need to be stored by the user's *devices*. Every single RADIUS capable client can store the password within the device. This creates the association of "the user logs into the device, the device then authenticates to the user to the network on the users behalf.". The user should *not* need to remember the RADIUS password, it should be up to the device. 

I hope this help explains the design and the reasons for the design choices we have made.

**Reply by kokx** [reply]
Is there a plan to support generating the RADIUS credential from the webinterface? So users can generate the password without needing to have a logged in device?

**Reply by yaleman** [reply]
Yep, everything's planned, happy to take a PR if someone wants to take it on.

**Reply by Gorian** [reply]
Just following up, because google brought me to this discussion.

As of May 2026, RADIUS credentials can be generated entirely in the WebUI
